shuup/core/api/basket.py

```diff
@@ -543,6 +546,26 @@ def _handle_setting_address(self, request, attr_field):
             data = BasketSerializer(request.basket, context=self.get_serializer_context()).data
             return Response(data, status=status.HTTP_200_OK)
 
+    @schema_serializer_class(ExtraDataSerializer)
+    @detail_route(methods=['post'])
+    def set_extra_data(self, request, *args, **kwargs):
+        self.process_request()
+        serializer = ExtraDataSerializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+        if serializer.is_valid():
+            if not serializer.validated_data:
```

Comment:
so if the dictionary is blank it is wrong? maybe this check should be done in the [serialize.validate()](http://www.django-rest-framework.org/api-guide/serializers/#object-level-validation) method